src/aatoolbox/datasources/chirps/chirps.py

```diff
@@ -0,0 +1,631 @@
+            If None, it is set to 1. Argument ignored when downloading
+            monthly data.
+        end_day: int, default = None
+            Data will be loaded up to day `end_day`.
+            If None, it is set to the last day of the month if year and month
+            were specified, otherwise to the day of the month of the year
+            for which most recent data is available.
+            Argument ignored when downloading monthly data.
+
+        Returns
+        -------
+        The processed CHIRPS dataset.
+        """
+        if self._frequency == "monthly":
+            start_day = None
+            end_day = None
+
+        # If no arguments are specified, all data in the processed data folder
+        # (with the appropriate resolution, frequency and corresponding to the
+        # chosen location) will be loaded
+        if all(
+            [
+                x is None
+                for x in [
+                    start_year,
+                    end_year,
+                    start_month,
+                    end_month,
+                    start_day,
+                    end_day,
+                ]
+            ]
+        ):
+            raw_path = self._get_raw_path(year=None, month=None, day=None)
+            processed_path = self._get_processed_path(raw_path)
+            filepath_list = list(
+                processed_path.parents[0].glob(processed_path.name)
+            )
+        # otherwise data loaded according to the range given as an input
+        else:
+            date_list, _ = self._create_date_list(
+                start_year=start_year,
+                end_year=end_year,
+                start_month=start_month,
+                end_month=end_month,
+                start_day=start_day,
+                end_day=end_day,
+            )
+            if self._frequency == "monthly":
+                filepath_list = [
+                    self._get_raw_path(year=year, month=month, day=None)
+                    for (year, month) in date_list
+                ]
+            else:
+                filepath_list = [
+                    self._get_raw_path(year=year, month=month, day=day)
+                    for (year, month, day) in date_list
+                ]
+        filepath_list.sort()
+
+        # Merge all files in one dataset
+        if len(filepath_list) == 0:
+            raise FileNotFoundError(
+                f"Cannot find any netcdf file in the form"
+                f"{processed_path}. Make "
+                f"sure that you have already called the 'process' method."
+            )
+        else:
+            try:
+                ds = xr.open_mfdataset(
+                    filepath_list,
+                    decode_times=False,
+                )
+                # include the names of all files that are included in the ds
+                ds.attrs["included_files"] = [f.stem for f in filepath_list]
+            except FileNotFoundError:
```

Comment:
Not sure what you mean. It can happen that some of the files are not existing (as the list filepath_list can be generated based on dates), so that is why I added a `FileNotFoundError`. I don't think there can be invalid nc files, as the user can only download valid ones (due to the check of the last date).

Reply:
About the first point: I think that's exactly what the check with `FileNotFoundError` does.

About the second point: I don't think this can happen at the loading state, as the downloaded datasets have already been opened once. However, I will add this check to the processing method.